examples/usual/example41.rb

```diff
@@ -0,0 +1,54 @@
+# frozen_string_literal: true
+
+module Usual
+  class Example41Transaction
+    def self.transaction(&block)
+      yield if block
+    end
+  end
+
+  class Example41 < ApplicationService::Base
+    output :number, type: Integer
+
+    make :assign_number_4
+
+    stage do
+      wrap_in ->(methods:) { Example41Transaction.transaction { methods } }
+      rollback :method_for_rollback
+
+      make :assign_number_5
+      make :assign_number_6
+      make :assign_number_7
+    end
+
+    make :assign_number_8
+
+    private
+
+    def assign_number_4
+      self.number = 4
+    end
+
+    def assign_number_5
+      self.number = 5
+    end
+
+    def assign_number_6
+      # self.number = 6
+
+      raise "bad number"
+    end
+
+    def assign_number_7
+      self.number = 7
+    end
+
+    def assign_number_8
+      self.number = 8
+    end
+
+    def method_for_rollback(_e)
+      self.number = 9
```

Comment:
Need to add an example with `fail!`.

Reply:
Done in https://github.com/afuno/servactory/pull/58/commits/a8ef8025460dab0f1161cfa3f2e61931c00dca03